### PR TITLE
feat(appmesh): add mesh owner property

### PR DESCRIPTION
**Pull request by Seiya6329**
**This PR is for internal-review purpose only**

#### REV:
- Adding `meshOwner` attribute to `IMesh`.
- Adding `meshOwner` property for resources

#### Design Note:
- Allowing to provide Mesh ARN from different AWS IAM account ID for `IMesh.fromMeshArn()` to enable working with shared mesh.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-appmesh/test/integ.mesh.expected.json [hunk 1]
```diff
@@ -95,15 +95,15 @@
     "vpcPublicSubnet1NATGateway9C16659E": {
       "Type": "AWS::EC2::NatGateway",
       "Properties": {
+        "SubnetId": {
```
Comment (rishijatia) [inline packages/@aws-cdk/aws-appmesh/test/integ.mesh.expected.json:98]: This seems unrelated to the PR. Can we revert this?

Comment (Seiya6329) [reply packages/@aws-cdk/aws-appmesh/test/integ.mesh.expected.json:98]: This actually looks like just switching the position for `SubnetId` and `AllocationId`.

Please be noted `integ.mesh.expcted.json` file is generated automatically after running `yarn integ integ.mesh.js` command.

**Comment by dfezzie** [comment]
As mentioned offline, I believe it would be better to model this as a property of the `IMesh`. If we are able to define the `meshOwner` property on the `IMesh` interface, we can simply use this to populate the `meshOwner` field in all the subresources without the end customer having to worry about it.

**Comment by Seiya6329** [comment]
@dfezzie - Thanks for the great advise! Adding the `meshOwner` attribute to `IMesh` truly makes everything work well!

File: packages/@aws-cdk/aws-appmesh/lib/mesh.ts [hunk 2]
```diff
@@ -40,6 +40,13 @@ export interface IMesh extends cdk.IResource {
    */
   readonly meshArn: string;
 
+  /**
+   * The AWS IAM account ID of the AppMesh mesh
```
Comment (dfezzie) [inline packages/@aws-cdk/aws-appmesh/lib/mesh.ts:44]: Small nit: We could include an `@see https://docs.aws.amazon.com/app-mesh/latest/userguide/sharing.html` here. I don't know if we do that in many other places, but could help customers

**Comment by dfezzie** [comment]
Yeah, I like this approach. Customers only need to have the mesh shared with them for this to work properly. Otherwise it just works!

**Review by dfezzie — APPROVED** [review]

File: packages/@aws-cdk/aws-appmesh/lib/mesh.ts (hunk 2 above)
Comment (Seiya6329) [reply packages/@aws-cdk/aws-appmesh/lib/mesh.ts:44]: Well noted. I will add the link.
